src/squlearn/util/executor.py

```diff
@@ -219,6 +257,16 @@ def __init__(
         if execution is None and backend is not None:
             # Only backend is given
             execution = backend
+
+        self._quantum_framework = "qiskit"
+        self._pennylane_device = None
+
+        if execution == "qiskit":
+            if shots is None:
+                execution = "statevector_simulator"
+            else:
+                execution = "qasm_simulator"
+
         if isinstance(execution, str):
```

Comment:
I think we should use elif
```suggestion
        elif isinstance(execution, str):
```

Reply:
done, moved "qiskit" keyword down